.github/workflows/circleci-trigger.yml

```diff
@@ -0,0 +1,89 @@
+name: CircleCI tests trigger
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches:
+      - main
+
+env:
+  PYTHONUNBUFFERED: 1
+  FORCE_COLOR: 1
+
+jobs:
+  circleci-trigger:
+    name: CircleCI tests trigger
+    environment: circleci_secrets
+    runs-on: ubuntu-latest
+    steps:
+      - name: Ping CircleCI pipeline
+        run: |
+          if [[ "${{ github.ref_type }}" == "tag" ]]; then
+            github_tag=${{ github.ref_name }}
+          else github_tag=""
+          fi
+
+          CREATED_PIPELINE=$(curl --request POST \
+            --url "${{ secrets.QA_PIPELINE_URL }}" \
+            --header "Circle-Token: ${{ secrets.CIRCLECI_API_KEY }}" \
+            --header "content-type: application/json" \
+            --data '{"branch":"${{ github.head_ref }}", "parameters":{"branch":"${{ github.head_ref }}", "vizro_branch": "${{ github.head_ref }}", "tag": "$github_tag"}}' \
+          | jq -r '.id'
+          )
+          if [[ "$CREATED_PIPELINE" == "null" ]]; then
+            CREATED_PIPELINE=$(curl --request POST \
+            --url "${{ secrets.QA_PIPELINE_URL }}" \
+            --header "Circle-Token: ${{ secrets.CIRCLECI_API_KEY }}" \
+            --header "content-type: application/json" \
+            --data '{"branch":"main", "parameters":{"branch":"main", "vizro_branch": "${{ github.head_ref }}", "tag": "$github_tag"}}' \
+            | jq -r '.id'
```

Comment:
What's the different between these two? One is triggered on merge to `main` and one is in the PR itself?

Reply:
ok, makes sense. Please could you add some comments to explain this? And maybe refactor also to make it clearer what's happening. The JSON payload is also pretty hard to read here - since we use `jq` to parse JSON already, maybe we can also use it to generate the JSON? Something like this, again mainly written by ChatGPT...

```
create_circleci_pipeline() {
  local branch=$1
  local vizro_branch=$2
  local tag=$3

  local json_data=$(jq -n --arg branch "$branch" --arg vizro_branch "$vizro_branch" --arg tag "$tag" '{branch: $branch, parameters: {branch: $branch, vizro_branch: $vizro_branch, tag: $tag}}')

  curl --request POST \
    --url "${{ secrets.QA_PIPELINE_URL }}" \
    --header "Circle-Token: ${{ secrets.CIRCLECI_API_KEY }}" \
    --header "content-type: application/json" \
    --data "$json_data" \
  | jq -r '.id'
}

CREATED_PIPELINE=$(create_circleci_pipeline "${{ github.head_ref }}" "${{ github.head_ref }}" "$github_tag")

if [[ "$CREATED_PIPELINE" == "null" ]]; then
  CREATED_PIPELINE=$(create_circleci_pipeline "main" "${{ github.head_ref }}" "$github_tag")
fi
```

Reply:
Nice suggestion! Thanks! Did it